include/Board.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <memory>
+#include <map>
+#include <vector>
+#include <iostream>
+
+#include "PieceFactory.h"
+
+class Piece;
+
+class Board {
+public:
+    static Board& getInstance();
+    int move(const std::string&);
+    std::pair<std::pair<int, int>, std::pair<int, int>> parseMoveString(const std::string&) const;
+    bool isValidPosition(int, int) const;
+    void fillBoard(const std::string&);
```

Comment:
Improve function naming and parameter names: Use descriptive function and parameter names to make the code more self-explanatory. For example, fillBoard could be renamed to initializeBoard or resetBoard.

Reply:
Done.